### ISSUE TITLE
Can't make it work...

### ISSUE THREAD

**Issue by LFranck** [upwork/phystrix#27]
Hi,

I'm using Phystrix to call a web service to print something on a web page.

Each time I call my page which the extern web service unavalaible:
- the **HealthCountsSnapshot** values never change
- so the **circuit is never open**
- for each call the result of **allowRequest** is always TRUE

What I understand from the code :
- the **values are never store** in (apc  is for statistic purpose only) so the next request don't know what happened before
- the code is the same than Hystrix (java) and that is the issue, the java thread stay in memory and so the values, **not in php**.

Is that a correct analysis?

Maybe I'm using Phystrix badly.

I'm waiting for your answer.

Best regards
FL

**Comment by persevereVon** [comment]
Well, same problem. 
https://github.com/upwork/phystrix/blob/c8ac8984e3ac754e733ac6cf3f109eeb75bf66e9/library/Odesk/Phystrix/MetricsCounter.php#L111

Here, each executes command has a new timestamp, so when get function execute, It must be 0. Maybe I have made some mistake.

Thank you for your effort.

**Comment by lcf** [comment]
Hi, the code is not the same as Java's code. In fact, most of the work behind Phystrix was assessing tools and technologies available to make the concept work in PHP environment.

APC is used to store the metrics and they are preserved between requests, that is the point. Only when the command is considered unhealthy the circuit is open. We have been using Phystrix in high load enterprise applications for a few years now.

I hope this helps. @LFranck If you can provide steps to reproduce the issue you're having, I would take take a look.

@persevereVon could you rephrase your comment, or better create a separate issue

**Comment by LFranck** [comment]
Hi,

I'm using ab (apache benchmark) to call the page 1000 times without connection.

I see : 
- the fallback is called
- but the counter never change
- allowRequest return always true
- the circuit is never Open

Please, tell me where in the code the values of HealthCountsSnapshot are saved for the next php thread.

If not, how other php thread work ?

best regards
FL

**Comment by persevereVon** [comment]
@lcf sorry for my English. And I have a question, if I change the cache engine to redis, is it still working properly? thank you!

**Comment by LFranck** [comment]
@persevereVon 
Can you create your own issue and not disturb mine, please ?

**Comment by persevereVon** [comment]
@LFranck OK, sorry

**Comment by lcf** [comment]
Cleared up the thread a little. @LFranck when execute sees a failure, it will call markFailure method which will increment value in APC in a tread safe manner, so all failures are counted.
Data in APC will be available until the bucket expires.

**Comment by LFranck** [comment]
I FOUND IT ! I FOUND IT !!!

Just here in odesk\phystrix-bundle\Resources\config\config.yml
By default : 
  phystrix.state_storage.class: Odesk\Phystrix\**ArrayStateStorage**

And I need : 
  phystrix.state_storage.class: Odesk\Phystrix\**ApcStateStorage**

But now, I have an another issue : my apache work with OPcache
and I read than both can't be installed in same time.

thanks
FL

**Comment by LFranck** [comment]
End of the story :
with the correct configuration and apcu installed, everything is fine.

Thanks for your help.
FL

**Comment by lcf** [comment]
I believe that's actually a mistake in the bundle component. Array adapter is only to be used for testing and development environment. I don't think it should be the default.

I'll look into it, thank you @LFranck!
